vundo.el

```diff
@@ -832,21 +850,28 @@ This function modifies `vundo--prev-mod-list',
           ;; Build tree.
           (vundo--build-tree mod-list mod-hash
                              (length vundo--prev-mod-list))))
+      
+      ;; Update cache.
+      (setq vundo--prev-mod-list mod-list
+            vundo--prev-mod-hash mod-hash
+            vundo--prev-undo-list undo-list
+            vundo--orig-buffer orig-buffer)
+      
+      ;; Record timestamps
+      (vundo--record-timestamps mod-list)
```

Comment:
Let's make `vundo--record-timestamps` more functional. By that I mean make it take both a mod-list and a timestamp list, and return a new timestamp list. Then here we write

```elisp
(setq vundo-timestamps (vundo--record-timestamps mod-list vundo-timestamps))
```

Similarly, pass the timestamp to `vundo--highlight-last-saved-node`. This way it's more coherent with existing code.

Reply:
I adopted this usage.  Note that the first thing `record-timestamps` does is nil out the list, so passing it in isn't necessary.  I also updated my use of time comparisons to the faster `time-less-p`.   Please give a quick test before merging (checked out here, but when you change the order of time comparisons it's worth verifying).